orbit-2-core/README.md

```diff
@@ -0,0 +1,319 @@
+yourself with the [overview documentation](docs/orbit-2-overview.md) before
+reading this section.
+
+The core module provides all the basic parts of Orbit:
+
+- Public API
+- Core DSL
+- Plugin registry
+- Container implementation
+
+## Architecture
+
+![Orbit 2 overview 1](docs/orbit-2-overview-1.svg)
+
+This diagram shows a simple representation of how an Orbit system (or similar
+systems like MVI/Redux/Cycle) works in simple principles.
+
+1. The UI sends actions asynchronously to some business component
+1. The business component applies some business logic to the incoming events
+1. The business component then emits some events further down the chain
+1. Every event is reduced with the current state of the system to produce a new
+   state
+1. The state is then emitted back to the UI which renders itself based upon
+   information within
+
+The main thing to remember is that the UI cannot make any business decisions
+by itself. It should know only how to render itself based on the input state.
+
+### Orbit concepts
+
+![Orbit 2 overview 2](docs/orbit-2-overview-2.svg)
+
+We can map the above logic onto real components.
+
+1. UI invokes functions on a class implementing the `ContainerHost` interface
+1. The functions call through to a `Container` instance through the `orbit`
+   DSL function
+1. Transformers apply business logic that transforms function parameters into
+   single or multiple events
+1. The reducer reduces the current state of the system with the incoming events
+   to produce new states
+1. The new state is sent to observers
+
+Notes:
+
+- UI can be an `Activity`, `Fragment` or a simple `View`. An Orbit system can
+  run even without any UI at all! Think of it as a background service.
+- Orbit operators are applied within the `orbit` DSL function block
+- All Orbit operators are optional.
+
+### Side effects
+
+In the real world such a system cannot exist with side effects. For example,
+navigation between screens can be considered a side effect. As such there's a
+third Orbit operator that can deal with side effects.
+
+![Orbit 2 overview 3](docs/orbit-2-overview-3.svg)
+
+The UI does not have to be aware of all side effects (e.g. why should the UI
+care if you send analytics events?). As such you can have side effects that do
+not post any event back to the UI.
+
+## Including the module
+
+Orbit 2 is a modular framework. You will need this module to get started!
+
+Additional functionality is provided through plugins included in separate
+modules. Install only what you need!
+
+```kotlin
+implementation("com.babylon.orbit2:orbit-core:<latest-version>")
+```
+
+## Orbit container
+
+A [Container](src/main/java/com/babylon/orbit2/Container.kt) is the heart of
+the Orbit MVI system. It retains the state, allows you to listen to side effect
+and state updates and allows you to modify the state through the `orbit`
+function which executes DSL operators of your desired business logic.
```

Comment:
elsewhere you refer to `DSL operators` as `Orbit operators`, I think I prefer the latter as I'm not sure everyone will understand what a DSL is. There are a few instances of this within the text we should tidy up